src/parquet/column/writer.h

```diff
@@ -189,6 +190,22 @@ inline void TypedColumnWriter<DType>::WriteBatch(int64_t num_values,
   if (current_encoder_->EstimatedDataEncodedSize() >= properties_->data_pagesize()) {
     AddDataPage();
   }
+  if (has_dictionary_ && !fallback_) { VerifyDictionaryFallback(); }
+}
+
+template <typename DType>
+inline void TypedColumnWriter<DType>::WriteBatch(int64_t num_values,
```

Comment:
We check for `DataPage` limits only after we have inserted the values. If a user writes a large number of values, the `DataPage` size can become much above the limit. The purpose of this chunking is to bound this. Even if a user writes large number of values, the chunking will ensure the `AddDataPage()` is called at a reasonable pagesize limit (above the default limit).